api/rosetta/errors.go

```diff
@@ -317,24 +325,35 @@ func formatError(err error) *echo.HTTPError {
 	if errors.As(err, &ibErr) {
 		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(ibErr.Description.Text,
 			withDetail("want_length", ibErr.WantLength),
+			withDetail("have_length", ibErr.HaveLength),
 		))
 	}
 	var iaErr failure.InvalidAccountAddress
 	if errors.As(err, &iaErr) {
 		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(iaErr.Description.Text,
 			withDetail("want_length", iaErr.WantLength),
+			withDetail("have_length", iaErr.HaveLength),
 		))
 	}
 	var itErr failure.InvalidTransactionHash
 	if errors.As(err, &itErr) {
 		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(itErr.Description.Text,
 			withDetail("want_length", itErr.WantLength),
+			withDetail("have_length", itErr.HaveLength),
 		))
 	}
 	var icErr failure.IncompleteBlock
 	if errors.As(err, &icErr) {
 		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(icErr.Description.Text))
 	}
+	var inErr failure.InvalidNetwork
+	if errors.As(err, &inErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidNetwork(inErr))
```

Comment:
This is perhaps a moot point since we discussed today about having all errors return a single HTTP status code. But to remain consistent with the previous version of the code, these two failures (misconfigured network or blockchain) return a `unprocessable entity` status code.

Even though it's not **exactly** a `bad request` error as some of the previous ones, I think it also fits in nicely, so perhaps these should just follow suite.

In any case, if they're going to be overridden, it makes no impact 😄

Reply:
I think this is correct. The only time we should have a `StatusBadRequest` is if the network or the blockchain are missing from the network identifier.